mcx/jax.py

```diff
@@ -71,3 +74,54 @@ def unravel_list(arr):
     )
 
     return flat, unravel_list
+
+
+def progress_bar_factory(num_samples):
+    """Factory that builds a progress bar decorator along
+    with the `set_tqdm_description` and `close_tqdm` functions
+    """
+
+    tqdm_pbar = tqdm(range(num_samples))
+
+    def set_tqdm_description(message):
+        tqdm_pbar.set_description(
+            message,
+            refresh=False,
+        )
+
+    def close_tqdm():
+        tqdm_pbar.close()
+
+    def _update_tqdm(arg, transform):
```

Comment:
This wouldn't go since we need it to discard the second argument in `id_tap`. Nitpick: `_update_tqdm(arg, _)`